__tests__/app.test.js

```diff
@@ -254,7 +254,56 @@ describe("POST /api/articles/:article_id/comments", () => {
       .send(newComment)
       .expect(400)
       .then(({ body }) => {
-        expect(body.msg).toBe("Bad Request - Invalid Data");
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  it("200: responds with the updated article", () => {
+    const newUpdate = { inc_votes: 10 };
+
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newUpdate)
+      .expect(200)
+      .then(({ body }) => {
+        const { updatedArticle } = body;
+
+        expect(updatedArticle).toMatchObject({
+          article_id: 1,
+          title: expect.any(String),
+          topic: "mitch",
+          author: "butter_bridge",
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: 110,
+          article_img_url: expect.any(String),
+        });
+      });
+  });
+
+  it("400: ERROR - responds with an error message when passed a malformed object", () => {
+    const newUpdate = {};
+
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newUpdate)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+
+  it("400: ERROR - responds with an error message when passed an object with invalid data type", () => {
+    const newUpdate = { inc_votes: "imAVote" };
+
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newUpdate)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
```

Comment:
Really good error handling, but dont forget that we also need to include error handling for when we are given an invalid article id and a valid article id but one that does not exist.